### PR TITLE
clippy updates for rust 1.48

### PR DESCRIPTION
**Pull request by bluejekyll**
Updates for clippy in 1.48

### CODE REVIEW

File: crates/resolver/src/lookup.rs [hunk 1]
```diff
@@ -664,36 +654,7 @@ pub mod tests {
...
-        );
-        assert_eq!(
-            lookup.next().unwrap(),
-            RData::A(Ipv4Addr::new(127, 0, 0, 2))
-        );
-        assert_eq!(lookup.next(), None);
-    }
-
-    #[test]
-    fn test_lookup_into_iter_vec() {
```
Comment (bluejekyll) [inline crates/resolver/src/lookup.rs:694]: This test was no longer relevant once the lookup iter was simplified.

**Comment by bluejekyll** [comment]
@djc, if you have a moment, I think this is a pretty small update. All new clippy warnings in 1.48.

File: crates/https/src/https_server.rs [hunk 2]
```diff
@@ -26,15 +26,15 @@ use crate::HttpsError;
...
+    this_server_name: Arc<str>,
     request: Request<R>,
 ) -> Result<BytesMut, HttpsError>
 where
     R: Stream<Item = Result<Bytes, h2::Error>> + 'static + Send + Debug + Unpin,
 {
     debug!("Received request: {:#?}", request);
 
-    let this_server_name: &String = this_server_name.borrow();
+    let this_server_name: &str = this_server_name.borrow();
```
Comment (djc) [inline crates/https/src/https_server.rs:37]: Can we just get rid of the type annotation here?

File: crates/resolver/src/name_server/name_server_pool.rs [hunk 3]
```diff
@@ -528,11 +528,11 @@ mod tests {
         let opts = ResolverOpts::default();
         let ns_config = { tcp };
         let name_server = NameServer::new_with_provider(ns_config, opts, conn_provider.clone());
-        let name_servers = Arc::new(vec![name_server]);
+        let name_servers: Arc<[_]> = Arc::from([name_server]);
```
Comment (djc) [inline crates/resolver/src/name_server/name_server_pool.rs:531]: Do we need this type annotation here? Might be a little nicer to write it as `Arc::<[_]>::from()` (which prevents repeating the `Arc` part at least.

File: crates/server/src/server/server_future.rs [hunk 4]
```diff
@@ -489,7 +489,7 @@ impl<T: RequestHandler> ServerFuture<T> {
         use trust_dns_rustls::tls_server;
 
         let spawner = Handle::current();
-        let dns_hostname = Arc::new(dns_hostname);
+        let dns_hostname: Arc<str> = Arc::from(dns_hostname);
```
Comment (djc) [inline crates/server/src/server/server_future.rs:492]: And same here... although I guess it's actually needed?

**Review by djc — COMMENTED** [review]
Mostly looks good, it's nice to have slices/strs instead of vecs/strings! I'm a little confused about the need for type annotations.

File: crates/https/src/https_server.rs (hunk 2 above)
Comment (bluejekyll) [reply crates/https/src/https_server.rs:37]: Oh, this one maybe.

File: crates/resolver/src/name_server/name_server_pool.rs (hunk 3 above)
Comment (bluejekyll) [reply crates/resolver/src/name_server/name_server_pool.rs:531]: The compiler gets confused on the from and to type, slice vs array.

File: crates/server/src/server/server_future.rs (hunk 4 above)
Comment (bluejekyll) [reply crates/server/src/server/server_future.rs:492]: This is just a problem of how late the actual type is bound, so the compiler was confused.

File: crates/resolver/src/name_server/name_server_pool.rs (hunk 3 above)
Comment (bluejekyll) [reply crates/resolver/src/name_server/name_server_pool.rs:531]: yeah, can't get rid of this.

**Comment by bluejekyll** [comment]
Ok, I got rid of one of them, the others are necessary, at least without additional refactoring.

**Review by djc — APPROVED** [review]

**Pull request merged**